curl_cffi/build.py

```diff
@@ -1,47 +1,76 @@
 import os
-import struct
 import platform
+import struct
 
 from cffi import FFI
 
-ffibuilder = FFI()
-# arch = "%s-%s" % (os.uname().sysname, os.uname().machine)
-uname = platform.uname()
 
+def abs_machine():
+    machine = platform.machine()
 
-if uname.system == "Windows":
-    if struct.calcsize("P") * 8 == 64:
-        libdir = "./lib64"
+    pointer_bits = struct.calcsize("P") * 8
+    if pointer_bits not in (32, 64):
+        raise Exception("Unsupported pointer size")
+
+    is_64 = pointer_bits == 64
+
+    # x86 based archs
+    if machine in ('AMD64', 'x86_64', 'i686'):
+        return "x86_64" if is_64 else "i686"
+    # arm based archs
+    elif machine in ('aarch64', 'arm64', 'armv6l', 'armv7l'):
+        return "aarch64" if is_64 else "arm"
     else:
+        raise Exception("Unsupported processor")
+
+
+ffibuilder = FFI()
+system = platform.system()
+machine = abs_machine()
+parent_dir = os.path.dirname(os.path.dirname(__file__))
+
+if system == "Windows":
+    if machine == "x86_64":
         libdir = "./lib32"
-elif uname.system == "Darwin":
-    libdir = "/Users/runner/work/_temp/install/lib"
+    elif machine == "i686":
+        libdir = "./lib64"
+    else:
+        libdir = "ERROR"
+elif system == "Darwin":
+    if machine in ("x86_64", "aarch64"):
```

Comment:
no need to detect here, there are no other archs for macOS

Reply:
return type of `abs_machine` is `Literal['x86_64', 'i686', 'aarch64', 'arm']`. let's be explicit on handling what it returns.